### ISSUE TITLE
Open up CE-Symm API so alternative visualization layers can be used.

### ISSUE THREAD

**Issue by andreasprlic** [biojava/biojava#393]
Currently the CE-symm visualization is tied closely to Jmol. Add a couple of utility classes that make it easier to build up a visualization also with other viewers.

**Comment by lafita** [comment]
What is exactly the idea you have in mind? I propose two methods for coloring the subunits:

1. Get a List of ResidueRanges for each protodomain (symmetric subunit), so that the viewer can color those parts entirely (already implemented in my branch).
2. Get a List of Lists with all ResidueNumbers for each protodomain, so that the viewer can color individually the equivalent residues (currently what Jmol does).

If we want to transfer also the display of the alignment we should think how to generalize all other alignment viewers (MultipleAlignment and StructureAlignment).

**Comment by andreasprlic** [comment]
I agree to everything you say. I need both 1) and 2) plus in general we should try to generalize alignment viewers so we can use others such as PV or NGL.

**Comment by lafita** [comment]
I created a new issue for StructureAlignment visualization #396, so that we have two threads about molecular visualization: one for symmetry (which should include the quaternary symmetry) and one for structural alignments.

**Comment by lafita** [comment]
I assign this issue to the 5.0 milestone because it is a major API change that will not happen before 4.2 release.
